osf/models/mixins.py

```diff
@@ -2214,6 +2215,21 @@ def expand_registration_responses(self):
             self.file_storage_resource,
         )
 
+    def copy_registration_responses_into_schema_response(self, draft_registration):
+        """Copies registration metadata into schema responses"""
+        from osf.models.schema_response import SchemaResponse
+        schema_response = SchemaResponse.create_initial_response(
+            self.creator,
+            self,
+            self.registration_schema
+        )
+        self.registration_responses = draft_registration.registration_responses
+        schema_response.update_responses(
+            self.registration_responses
+        )
+        schema_response.reviews_state = ApprovalStates.APPROVED.db_name
```

Comment:
Letting this be APPROVED will let users POST a new SchemaResponse before the Registration is APPROVED. Thus the suggestion to update state for the initial response as part of `update_moderation_state`, otherwise we need to make the SchemaResponse aware of its parent's state in ways that feel not good.